docs/Gopkg.toml.md

```diff
@@ -0,0 +1,105 @@
+ ```toml
+required = ["github.com/user/thing/cmd/thing"]
+```
+
+## `ignored`
+ignored lists a set of packages (not projects) that are ignored when dep statically analyzes source code. Ignored packages can be in this project, or in a dependency.
+```toml
+ignored = ["github.com/user/project/badpkg"]
+```
+
+## `[metadata]`
+`metadata` can exist at the root as well as under `constraint` and `override`. 
+At the root `metadata` defines `metadata` about the project that could be used by other independent systems. 
+
+`metadata` under `constraint` or `override`  defines `metadata` about the dependency or override that could be used by other independent systems. The `metadata` defined here will be ignored by dep.
+```toml
+[metadata]
+key1 = "value that convey data to other systems"
+system1-data = "value that is used by a system"
+system2-data = "value that is used by another system"
+```
+
+## `[[constraint]]`
+constraints are rules for how directly imported projects may be incorporated into the depgraph. They are respected by dep whether coming from the Gopkg.toml of the current project or a dependency.
+```toml
+[[constraint]]
+  # Required: the root import path of the project being constrained.
+  name = "github.com/user/project"
+  # Recommended: the version constraint to enforce for the project.
+  # Only one of "branch", "version" or "revision" can be specified.
+  version = "1.0.0"
+  branch = "master"
+  revision = "abc123"
+
+  # Optional: an alternate location (URL or import path) for the project's source.
+  source = "https://github.com/myfork/package.git"
+
+  # Optional: metadata about the dependency or override that could be used by other independent systems
+  [metadata]
+  key1 = "value that convey data to other systems"
+  system1-data = "value that is used by a system"
+  system2-data = "value that is used by another system"
+  ```
+
+## `[[override]]`
+ Overrides have the same structure as `[[constraint]]`, but supersede all `[[constraint]]` declarations from all projects. Only `[[override]]` from the current project's are applied.
+
+ **Overrides are a sledgehammer. Use them only as a last resort.**
+```toml
+[[override]]
+  # Required: the root import path of the project being constrained.
+  name = "github.com/user/project"
+  # Optional: specifying a version constraint override will cause all other
+  # constraints on this project to be ignored; only the overridden constraint need be #satisfied. Again, only one of "branch", "version" or "revision" can be specified.
+  version = "1.0.0"
+  branch = "master"
+  revision = "abc123"
+  # Optional: specifying an alternate source location as an override will enforce that
+  # the alternate location is used for that project, regardless of what source 
+  # location any dependent projects specify.
+  source = "https://github.com/myfork/package.git"
+
+  # Optional: metadata about the dependency or override that could be used by other independent systems
```

Comment:
Optional: metadata about the constraint or override that could be used by other independent systems.